k8s/README.md

```diff
@@ -0,0 +1,262 @@
+[KubernetesPodOperator](https://airflow.readthedocs.io/en/stable/kubernetes.html#kubernetes-operator).
+
+**NB** This doesn't describe how to setup the 
+[KubernetesExecutor](https://airflow.readthedocs.io/en/stable/kubernetes.html#kubernetes-executor),
+which creates pods to execute other types of operators (e.g. Bash, Python).
+
+## Assumptions
+* Kubernetes is up and running (`systemctl status kubelet`)
+* You have `kubectl` installed and talking to the cluster
+* You are running these commands in this directory
+* You're using a NFS storage client with `StorageClassName='nfs-client'`
+* You're using traefik as your frontend service reverse proxy
+
+## 1. Namespace and Role
+First up, we need to handle the permissions end of things.
+
+1. Creating the namespaces, `airflow` (where the service will live), and
+`airflow-workers` (where the KubernetesPodOperator's pods will live, 
+albeit briefly): `kubectl apply -f resources/airflow_namespaces.yaml`
+
+Result:
+```
+namespace/airflow-workers created
+namespace/airflow created
+```
+
+2. Creating the Role, `airflow-role`. This role gives permission to mess 
+with pods in the `airflow-workers` namespace: `kubectl apply -f resources/airflow_role.yaml`
+
+Result:
+```
+role.rbac.authorization.k8s.io/airflow-role created
+```
+
+3. Binding the Role, `airflow-role`, to the `airflow` service account 
+(which doesn't exist yet): `kubectl apply -f resources/airflow_role_bind.yaml`
+
+Result:
+```
+rolebinding.rbac.authorization.k8s.io/airflow-role-bind created
+```
+
+The idea is that the airflow service account can **only** control pods
+in the `airflow-workers` namespace. This means even if Airflow gains
+sentience, goes rogue and tries to kill us all, it can only do so in the
+`airflow-workers` namespace.
+
+## 2. DAG and Log Persistence
+When deploying Airflow as a service, the `dags` and `logs` directories
+need to be shared between the various components.
+
+* Creating persistent volumes. You want to do this is if you want 
+control of exactly where your `dags` and `logs` directories live.
+Command: `kubectl apply -f resources/airflow_volumes.yaml`.
+
+**NB** Your NFS configuration is almost certainly different to ours. 
+You are probably going to have to change the `nfs.server` and `nfs.path`
+keys.
+
+**NBx2** if you decide to the above, you probably want to think carefully about why
+you want such fine-grained control.
+
+* Creating persistent volume claims. You want to do this so that your
+`dags` and `logs` directories persist between airflow deployments:
+`kubectl apply -f resources/airflow_volume_claims.yaml`
+
+Result:
+```
+persistentvolumeclaim/airflow-dags created
+persistentvolumeclaim/airflow-logs created
+```
+
+## 3. Setting Values
+There isn't anything to do here, other than review and change some of
+the values as required in [the supplied file](./resources/airflow-values.yaml).
+
+Of interest:
+* `airflow.fernetKey` - generate something strong for this.
```

Comment:
`echo $(openssl rand -hex 32)`

Reply:
"You really should generate a new fernet key for security purposes. You only need to change the others if you want to deviate from our defaults."

Reply:
sure